opencollective/opencollective-frontend
# Blockstack SignIn

This PR
* adds buttons for Blockstack authentication and user creation
* solves the frontend part of opencollective/opencollective#1749
* replaces https://github.com/opencollective/opencollective-frontend/pull/1833
* uses the email sign in for users with existing profiles for the email address associated with the blockstack id.

Some screenshots:
![Screenshot from 2019-05-22 15-06-14](https://user-images.githubusercontent.com/1449049/58216412-c464f980-7cfe-11e9-820b-129306f85dd7.png)
![Screenshot from 2019-06-19 16-56-35](https://user-images.githubusercontent.com/1449049/59776622-6ef71a80-92b3-11e9-9f72-0c3519db0d90.png)


Testing in local dev environment:
1. checkout opencollective-api branch feature/blockstack
1. install the api app dependencies (I needed to do an extra `npm install bcrypt` because of node version issue)
1. setup and migrate the database
1. run the api app
1. checkout this branch for frontend
1. install frontend app
`npm install`
1. in env.js replace WEBSITE_URL localhost:3000 with 127.0.0.1:3000 
1. run the frontend app
`npm run dev`
1. launch browser for frontend using 127.0.0.1:3000, if needed enable CORS browser plugin
1. Click "sign in"  button to use blockstack
1. When asked to open xdg-open, just cancel. Only happens for some browsers.
1. create a blockstack id following the wizzard, select the newly created blockstack id and continue to the opencollective site
1. see how the profile is created and the user logged in
1. logout to test signin
1. sign in with blockstack

(Screencast of early work)
![opencollective](https://user-images.githubusercontent.com/1449049/58478001-41f09580-8155-11e9-99c9-ccb5380f130f.gif)
